core/selection.cpp

```diff
@@ -222,6 +222,27 @@ bool setSelectionKeepCurrent(const std::vector<dive *> &selection)
 	return current_dive != oldCurrent;
 }
 
+void setTripSelection(dive_trip *trip, dive *currentDive)
+{
+	if (!trip)
+		return;
+
+	current_dive = currentDive;
```

Comment:
is there a need to check that this new `current_dive` is actually part of the trip?